src/main/java/impl/expression/Lexer.java

```diff
@@ -0,0 +1,56 @@
+package impl.expression;
+
+import api.expression.ParseException;
+import impl.expression.lexis.*;
+import impl.expression.lexis.Number;
+
+public class Lexer {
+
+    private String input;
+    private int currentPosition;
+
+    public Lexer(String input) {
+//        reader = new StringReader(input);
+//        Okay, StringReader seems not to support seek for some reason and is just cumbersome...
+//        W'll just store the string itself, maybe it's better to use CharArray tho...
+        this.input = input;
+    }
+
+    public Lexem getNextToken() throws ParseException {
+        if (currentPosition >= input.length()) {
+            return new Lexem(LexemType.EOE);
+        }
+        char symbol = input.charAt(currentPosition++);
+        switch (symbol) {
+            case '\t': //fallthrough
```

Comment:
https://en.wikipedia.org/wiki/Whitespace_character

Reply:
It's a little nod to `\r`? or a hint to use some `.isWhitespace()`?

Reply:
Hint to use `.isWhitespace()` :)